Source/WebCore/platform/graphics/cocoa/ImageUtilitiesCocoa.mm

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "ImageUtilities.h"
+
+#if PLATFORM(MAC)
+#import <pal/spi/mac/NSImageSPI.h>
+#else
+#import <pal/spi/cocoa/CoreSVGSPI.h>
+#endif
+
+#if !PLATFORM(MAC)
+SOFT_LINK_PRIVATE_FRAMEWORK(CoreSVG)
+SOFT_LINK(CoreSVG, CGSVGDocumentCreateFromData, CGSVGDocumentRef, (CFDataRef data, CFDictionaryRef options), (data, options))
+SOFT_LINK(CoreSVG, CGSVGDocumentRelease, void, (CGSVGDocumentRef document), (document))
+#endif
+
+namespace WebCore {
+
+bool canDataBeDecodedAsSVG(std::span<const uint8_t> data)
+{
+    RetainPtr nsData = toNSData(data);
+#if PLATFORM(MAC)
+    static Class NSSVGImageRep = NSClassFromString(@"_NSSVGImageRep");
+    if (!NSSVGImageRep)
+        return false;
+
+    return [NSSVGImageRep canInitWithData:nsData.get()];
+#else
+    RetainPtr parser = adoptNS([[NSXMLParser alloc] initWithData:nsData.get()]);
+    if (!parser || ![parser parse])
+        return false;
+
+    CGSVGDocumentRef document = CGSVGDocumentCreateFromData((__bridge CFDataRef)nsData.get(), nullptr);
```

Comment:
Do we need to use `CGSVGDocumentRelease` or can we just use `adoptCF` for this instead?

We have `bridge_cast`, which is better than writing `(__bridge CFDataRef)`.

Reply:
I updated the function to use `DOMParser` instead of CoreSVG as we may want to use WebKit's own parser for rendering image later, so this part is removed.